lib/home_page.dart

```diff
@@ -1,22 +1,511 @@
+  }
+
+  @override
+  void dispose() {
+    supabaseManager.supabaseClient.removeChannel(_messagesSubscriptions);
+    supabaseManager.supabaseClient.removeChannel(_rideEventsSubscriptions);
+    supabaseManager.supabaseClient.removeChannel(_ridesSubscriptions);
+    supabaseManager.supabaseClient.removeChannel(_drivesSubscriptions);
+    super.dispose();
+  }
+
+  Future<void> load() async {
+    final List<Model> items = <Model>[];
+    final int profileId = supabaseManager.currentProfile!.id!;
+    final List<Map<String, dynamic>> messagesData = parseHelper.parseListOfMaps(
+      await supabaseManager.supabaseClient.from('messages').select<List<Map<String, dynamic>>>('''
+      *,
+      sender: sender_id(*)
+      )
+    ''').eq('read', false).neq('sender_id', profileId).order('created_at'),
+    );
+    items.addAll(Message.fromJsonList(messagesData));
+    final List<Map<String, dynamic>> rideEventsData = parseHelper.parseListOfMaps(
+      await supabaseManager.supabaseClient.from('ride_events').select<List<Map<String, dynamic>>>('''
+      *,
+      ride: ride_id(*,
+        rider: rider_id(*),
+        drive: drive_id(*,
+          driver: driver_id(*)
+          )
+      )
+      ''').eq('read', false).order('created_at'),
+    );
+    items.addAll(RideEvent.fromJsonList(rideEventsData).where((RideEvent rideEvent) => rideEvent.isForCurrentUser()));
+    items.sort((Model a, Model b) => b.createdAt!.compareTo(a.createdAt!));
+    final DateTime now = DateTime.now();
+    final DateTime today = DateTime(now.year, now.month, now.day);
+    final DateTime tomorrow = today.add(const Duration(days: 2));
+    final List<Trip> trips = <Trip>[];
+    trips.addAll(
+      Ride.fromJsonList(
+        parseHelper.parseListOfMaps(
+          await supabaseManager.supabaseClient
+              .from('rides')
+              .select<List<Map<String, dynamic>>>()
+              .eq('rider_id', profileId)
+              .eq('status', RideStatus.approved.index)
+              .lt('start_time', tomorrow)
+              .gte('start_time', today),
+        ),
+      ),
+    );
+    trips.addAll(
+      Drive.fromJsonList(
+        parseHelper.parseListOfMaps(
+          await supabaseManager.supabaseClient
+              .from('drives')
+              .select<List<Map<String, dynamic>>>()
+              .eq('driver_id', profileId)
+              .eq('cancelled', false)
+              .lt('start_time', tomorrow)
+              .gte('start_time', today),
+        ),
+      ),
+    );
+    trips.sort((Trip a, Trip b) => a.startTime.compareTo(b.startTime));
+    _upcomingTripsCount = trips.length;
+    items.insertAll(0, trips);
+
+    setState(() {
+      _items = items;
+      _fullyLoaded = true;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       appBar: AppBar(
-        title: const Text('Home'),
+        title: Text('${S.of(context).hello} ${supabaseManager.currentProfile!.username} :)'),
```

Comment:
```suggestion
        // Wave emoji
        title: Text('${S.of(context).hello} ${supabaseManager.currentProfile!.username} \u{1F44B}'),
```

Reply:
Also, I feel like the emoji should be part of the translation. Not because we want to use other emojis in different languages per se, but I feel like it would be cleaner and more future-proof (Right-to-left languages?)
So, I'd add a username parameter to the hello text in the arb file and include the emoji there